conda_build/create_test.py

```diff
@@ -87,6 +88,28 @@ def create_shell_files(m):
                     f.write("if errorlevel 1 exit 1\n")
                 has_tests = True
 
+    libs = ensure_list(m.get_value('test/check_shared_libs_load', []))
+    if libs:
+        if sys.platform.startswith('win'):
+            lib_path = os.path.join(m.config.test_prefix, 'Library', 'bin')
+        else:
+            lib_path = os.path.join(m.config.test_prefix, 'lib')
+
+        with open(join(m.config.test_dir, name), 'a') as f:
+            f.write('\n\n')
+            for lib in libs:
+                lib_loc = os.path.join(lib_path, lib + get_shlib_ext())
+                # test existence
+                if sys.platform.startswith('win'):
+                    f.write('if not exist ' + lib_loc + ' exit 1\n')
+                else:
+                    f.write('test -f ' + lib_loc + '\n')
+                # test loading
+                f.write('python -c "import ctypes; ctypes.cdll[\'' + lib_loc + '\']"\n')
```

Comment:
It takes one line to do this with Python. Seems like a good use of Python to me. 😕

Admittedly there is this issue of loading the library with the wrong runtime on Windows. However wouldn't this be an issue even if Python weren't used? Whatever tool we need to use will still be loading the library as that is the goal. If it has the wrong runtime, won't this still be a problem? How do you envision avoiding this conundrum?

As to the compiler, it is my understanding we can't use the compiler on Windows in the test phase correctly. So this is a different issue. If we could use the compiler on Windows, that would be great and would help solve other problems. Then there is the question of what compiler we are using on the other platforms. TBH this sounds more challenging to me than just getting Python to work here. Though gains here would definitely be valuable in other problems, so not opposed to it.